frontend/catalyst/jax_tracer.py

```diff
@@ -840,6 +841,33 @@ def trace_quantum_measurements(
                     results = (jnp.asarray(results[0], jnp.int64), results[1])
                 out_classical_tracers.extend(results)
                 counts_tree = tree_structure(("keys", "counts"))
+
+                def change_child(leaf, sub_tree):
```

Comment:
Do these helper functions require any local variables in the `trace_quantum_measurement` function? If not then it would be better to not nest them inside.